internal/tracegen/config_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracegen
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+func Test_Run(t *testing.T) {
+	logger := zap.NewNop()
+	tp := sdktrace.NewTracerProvider()
+
+	tests := []struct {
+		name        string
+		config      *Config
+		expectedErr error
+	}{
+		{
+			name:        "Empty config",
+			config:      &Config{},
+			expectedErr: errors.New("either `traces` or `duration` must be greater than 0"),
+		},
+		{
+			name: "Non-empty config",
+			config: &Config{
+				Workers:    2,
+				Traces:     10,
+				ChildSpans: 5,
+				Attributes: 20,
+				AttrKeys:   50,
+				AttrValues: 100,
+				Duration:   time.Second,
+				Service:    "custom",
```

Comment:
do you need all these values for the test?

Reply:
Need most of them